src/engine/pieces/pawn.ts

```diff
@@ -1,13 +1,49 @@
 import Piece from './piece';
 import Player from '../player';
 import Board from '../board';
+import Square from '../square';
 
 export default class Pawn extends Piece {
     public constructor(player: Player) {
         super(player);
     }
 
     public getAvailableMoves(board: Board) {
-        return new Array(0);
+        const currentPosition = board.findPiece(this);
+        let moveList = [];
+        
+        if (this.player === Player.WHITE){
+            const candidateMove1 = Square.at(currentPosition.row+1, currentPosition.col);
+            if (!board.isPieceAtSquare(candidateMove1)){
+
+                moveList.push(Square.at(currentPosition.row+1, currentPosition.col));
+                
+                if (currentPosition.row === 1) {
+                    const candidateMove2 = Square.at(currentPosition.row+2, currentPosition.col);
+
+                    if (!board.isPieceAtSquare(candidateMove2)){
+                        moveList.push(Square.at(currentPosition.row+2, currentPosition.col));
+                        
+                    }
+                }
+            }
+            
+        } else{
+            const candidateMove3 = Square.at(currentPosition.row-1, currentPosition.col);
+            if (!board.isPieceAtSquare(candidateMove3)){
+
+                moveList.push(Square.at(currentPosition.row-1, currentPosition.col));
+                
+                if (currentPosition.row === 6) {
+                    const candidateMove4 = Square.at(currentPosition.row-2, currentPosition.col);
+
+                    if (!board.isPieceAtSquare(candidateMove4)){
+                        moveList.push(Square.at(currentPosition.row-2, currentPosition.col));
+                        
+                    }
+                }
+            }
+        }
+        return moveList;
```

Comment:
I think we can commonise this a bit to make it neater and easier to understand.

e.g. we could save the direction to move pieces and the start row as variables which we pull from what player it is and the can combine the code into a commonised function.

Breaking this up into functions would also help it be more clear what is happening